### PR TITLE
kubelet: revert contextual logging support

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind bug
/kind failing-test

#### What this PR does / why we need it:

It turned out that using testing.T for logging had a race condition and
potential panic because the tests keep goroutines running and testing.T is not
supposed to be used anymore after test
completion (https://github.com/kubernetes/kubernetes/issues/110854).

Either the code must be fixed to terminate all goroutines before a test
ends (seems non-trivial because the usage of goroutines is fairly complex in
the shutdown manager code), or ktesting must handle this case. A solution for
this is pending in https://github.com/kubernetes/klog/pull/337.

Either way, solving this will take a bit longer. In the meantime, we should
revert the change to get unit testing stable again.

#### Which issue(s) this PR fixes:

Related-to: https://github.com/kubernetes/kubernetes/issues/110854

#### Special notes for your reviewer:

In order to make this a local change, ktesting is kept as a dependency of the
test and thus Kubernetes.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go
Comment (kerthcet) [inline pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go:714]: Should we defer to revert this with `LogToStderr(true)`?

File: pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go
Comment (pohly) [reply pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go:671]: I am reverting to the code before my changes. It was like that.

File: pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go
Comment (pohly) [reply pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go:714]: Same here - I am just reverting.

**Comment by kerthcet** [comment]
Totally looks good to me, but would like to leave to someone more familiar with this.

**Comment by pohly** [comment]
Here's the diff against the code before PR #110504:

```diff
$ git diff 0669ba386bde2e756bc9c6779ad4a4f036200f28 -- pkg/kubelet/kubelet.go pkg/kubelet/kubelet_test.go pkg/kubelet/nodeshutdown 
diff --git a/pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go b/pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go
index e682c0963f0..e58f2d4dea1 100644
--- a/pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go
+++ b/pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go
@@ -45,6 +45,14 @@ import (
        probetest "k8s.io/kubernetes/pkg/kubelet/prober/testing"
        "k8s.io/utils/clock"
        testingclock "k8s.io/utils/clock/testing"
+
+       // https://github.com/kubernetes/kubernetes/pull/110504 started using
+       // ktesting for the first time in Kubernetes and therefore added it to
+       // "vendor". We need to revert that PR for a short while (see
+       // https://github.com/kubernetes/kubernetes/issues/110854) but to
+       // make the revert local to this directory, we keep importing that
+       // package (no change to vendor).
+       _ "k8s.io/klog/v2/ktesting/init"
 )
 
 // lock is to prevent systemDbus from being modified in the case of concurrency.
```

**Review by endocrimes — APPROVED** [review]
Simple revert that fixes the data race

/lgtm
/assign @mrunalp

**Comment by pohly** [comment]
/hold

Let's merge https://github.com/kubernetes/kubernetes/pull/111001 instead.

**Comment by pohly** [comment]
/close

The klog update was merged, the revert should not be needed anymore.

**Pull request closed without merging**
